pkgs/development/libraries/unixODBCDrivers/default.nix

```diff
@@ -181,6 +181,104 @@
+
+      versionMajor = "18";
+      versionMinor = "1";
+      versionAdditional = "1.1";
+      versionSuffix = lib.strings.optionalString stdenv.targetPlatform.isLinux "-1";
+
+    src =
+      if stdenv.targetPlatform.isDarwin
+      then
+        if stdenv.targetPlatform.isAarch64
+        then
+          fetchurl {
+            url = "https://download.microsoft.com/download/6/4/0/64006503-51e3-44f0-a6cd-a9b757d0d61b/${pname}-${version}-arm64.tar.gz";
+            sha256 = "116xl8r2apr5b48jnq6myj9fwqs88yccw5176yfyzh4534fznj5x";
+          }
+        else
+          fetchurl {
+            url = "https://download.microsoft.com/download/6/4/0/64006503-51e3-44f0-a6cd-a9b757d0d61b/${pname}-${version}-amd64.tar.gz";
+            sha256 = "1fn80byn1yihflznxcm9cpj42mpllnz54apnk9n46vzm2ng2lj6d";
+          }
+      else
+        if stdenv.targetPlatform.isAarch64
+        then
+          fetchurl {
+            url = "https://packages.microsoft.com/debian/11/prod/pool/main/m/${pname}/${pname}_${version}_arm64.deb";
+            sha256 = "0zphnbvkqdbkcv6lvv63p7pyl68h5bs2dy6vv44wm6bi89svms4a";
+          }
+        else
+          fetchurl {
+            url = "https://packages.microsoft.com/debian/11/prod/pool/main/m/${pname}/${pname}_${version}_amd64.deb";
+            sha256 = "1f0rmh1aynf1sqmjclbsyh2wz5jby0fixrwz71zp6impxpwvil52";
+          };
+
+    nativeBuildInputs =
+      if stdenv.targetPlatform.isDarwin
+      then
+        [
+          # Fix up the names encoded into the dylib, and make them absolute.
+          fixDarwinDylibNames
+        ]
+      else
+        [
+          dpkg
+          patchelf
+        ];
+
+      unpackPhase = lib.strings.optionalString stdenv.targetPlatform.isLinux "dpkg -x $src ./";
+
+      installPhase =
+        if stdenv.targetPlatform.isDarwin
+        then
+          ''
+            mkdir -p $out
+            tar xf $src --strip-components=1 -C $out
+          ''
+        else
+          ''
+            mkdir -p $out
+            mkdir -p $out/lib
+            cp -r opt/microsoft/msodbcsql${versionMajor}/lib64 opt/microsoft/msodbcsql${versionMajor}/share $out/
+          '';
+
+      # Replace the hard-coded paths in the dylib with nixpkgs equivalents.
+      fixupPhase = lib.strings.optionalString stdenv.targetPlatform.isDarwin ''
+        ${stdenv.cc.bintools.targetPrefix}install_name_tool \
+          -change /usr/lib/libiconv.2.dylib ${libiconv}/lib/libiconv.2.dylib \
+          -change /opt/homebrew/lib/libodbcinst.2.dylib ${unixODBC}/lib/libodbcinst.2.dylib \
+          $out/lib/libmsodbcsql.${versionMajor}.dylib
+      '';
+
+      postFixup = lib.strings.optionalString stdenv.targetPlatform.isLinux ''
+        patchelf --set-rpath ${lib.makeLibraryPath [ unixODBC openssl libkrb5 libuuid stdenv.cc.cc ]} \
+          $out/lib/libmsodbcsql-${versionMajor}.${versionMinor}.so.${versionAdditional}
+      '';
+
+      passthru = {
+        fancyName = "ODBC Driver ${versionMajor} for SQL Server";
```

Comment:
What is this needed for?

Reply:
It's useful when building upon it to generate the `odbcinst.ini` file. [Here's an example.](https://github.com/hasura/graphql-engine/blob/47baf4addc33f5f88ecc2309691418a6753a2b11/nix/shell.nix#L20-L32)

Reply:
> It's useful when building upon it to generate the `odbcinst.ini` file. [Here's an example.](https://github.com/hasura/graphql-engine/blob/47baf4addc33f5f88ecc2309691418a6753a2b11/nix/shell.nix#L20-L32)

It'd be nice to comment something near this, near the `passthru` value...

Reply:
It's the same explanation for all ODBC drivers, so I added a long comment to the top of the file and short comments near each `passthru`.